### PR TITLE
Assignment4 - Joshua Lester & Spencer West

### PR DESCRIPTION
**Pull request by Joshua-Lester3**
Joshua Lester: @Joshua-Lester3
Spencer West: @Saawest

URL: https://purple-plant-006657c1e.5.azurestaticapps.net/ 

Been working with Grant and Meg and there is an error when calling /game/result endpoint on the azure static web app. It works in localhost. They told me not to worry about fixing it as it was something they added in class, not part of the assignment.

### CODE REVIEW

File: wordle-web/pages/index.vue [hunk 1]
```diff
@@ -1,95 +1,80 @@
...
-        v-for="(guess, i) of game.guesses"
-        :key="i"
-        :guess="guess"
-      />
-
-      <div class="my-10">
-        <Keyboard />
-      </div>
+  <v-container fluid class="image-container">
+      <v-img src="../public/edited-vaporwave-landingPage.jpg">
```
Comment (timmytaters) [inline wordle-web/pages/index.vue:3]: I like the image, just gotta put it on the other pages as well

File: wordle-web/components/SettingsDialog.vue [hunk 2]
```diff
@@ -0,0 +1,78 @@
+<template>
+  <v-dialog v-model="modelValue">
+    <v-card class="ma-auto pa-4" min-width="300">
```
Comment (timmytaters) [inline wordle-web/components/SettingsDialog.vue]: I believe it's this pa-4 making it transparent

File: Wordle.Api/Wordle.Api/Models/Player.cs [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+namespace Wordle.Api.Models;
+
+public class Player
+{
+    public int PlayerId { get; set; }
+    [Required]
+    public required string Name { get; set; }
+    public int GameCount { get; set; }
+    public double AverageAttempts { get; set; }
+    public double AverageSeconds { get; set; }
```
Comment (timmytaters) [inline Wordle.Api/Wordle.Api/Models/Player.cs:12]: I believe average seconds was supposed to be int

File: wordle-web/components/SettingsDialog.vue [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+      <v-sheet class="ml-6" width="200">
+        <v-form @submit.prevent>
+          <v-select
+            color="secondary"
+            v-model="selectedTheme"
+            label="Select theme"
+            :items="['Regular', 'Jungle', 'Flamingo', 'Whurdle']"></v-select>
+          <v-btn
+            color="secondary"
+            @click="changeColorScheme"
```
Comment (timmytaters) [inline wordle-web/components/SettingsDialog.vue:26]: Could have this also toggle ShowSettingsDialog from default.vue

**Review by timmytaters — COMMENTED** [review]
It seems like there's a lot of issues unfortunately
- The main and random game just go to an infinite loading screen when won or lost
- The leaderboard and last 10 words are empty, so they're hypothetically not getting posted and/or displayed
- Valid words list doesn't update, and should probably close when a word is clicked and filled in
- The submit button on the theme doesn't close the window, you have to click off
- The top right "how to play section" and the theme selection probably shouldn't be transparent, makes it hard to read

Features
Make all pages responsive ✅
1080p desktop ✅
iPad Air ✅
Samsung Galaxy S20 Ultra Super Mega Excellent ✅
IPhone SE ✅
Create a improved landing page and site theme (Background, make it look attractive to play) ✅
Theme should extend all pages ❌
Add an instructions page that is available via the hamburger menu that details the features of the game and how to play ✅
Create a new Wordle game page that reuses components but will provide the same word for a specific day. Everyone going to the page that day will get the same word. The page should indicate Wordle of the Day or something like that ❌
Create a page that lists the last 10 daily words with statistics. Number of plays, Average score, Average time. (just list the date, don't show the word silly) ❌
It should indicate if the user has played that day ❌
Clicking on the item should take the player to that game (Extra Credit) ❌

**Review by rahat15 — COMMENTED** [review]
great work!!

Features
Make all pages responsive ✅
1080p desktop ✅
iPad Air ✅
Samsung Galaxy S20 Ultra Super Mega Excellent ✅
IPhone SE ✅
Create a improved landing page and site theme (Background, make it look attractive to play) ✅
Theme should extend all pages ✅
Add an instructions page that is available via the hamburger menu that details the features of the game and how to play ✅
Create a new Wordle game page that reuses components but will provide the same word for a specific day. Everyone going to the page that day will get the same word. The page should indicate Wordle of the Day or something like that ✅
Create a page that lists the last 10 daily words with statistics. Number of plays, Average score, Average time. (just list the date, don't show the word silly) ❌
It should indicate if the user has played that day ❌
Clicking on the item should take the player to that game (Extra Credit) ❌

File: wordle-web/pages/index.vue (hunk 1 above)
Comment (Joshua-Lester3) [reply wordle-web/pages/index.vue:3]: Partner took care of it! :)

File: wordle-web/components/SettingsDialog.vue (hunk 2 above)
Comment (Joshua-Lester3) [reply wordle-web/components/SettingsDialog.vue]: Thank you!

File: Wordle.Api/Wordle.Api/Models/Player.cs (hunk 3 above)
Comment (Joshua-Lester3) [reply Wordle.Api/Wordle.Api/Models/Player.cs:12]: Thanks for the note. Double makes more sense to me for an average, but I could be wrong.

File: wordle-web/components/SettingsDialog.vue (hunk 4 above)
Comment (Joshua-Lester3) [reply wordle-web/components/SettingsDialog.vue:26]: I think submitting the color theme shouldn't close the settings dialog, as the user might want to change the dark/light mode. Thank you for the suggestion though :)

**Review by vicdotexe — COMMENTED** [review]
- Make all pages responsive ❌✅ `Content on smaller mobile devices was squished (last 10 words page), and the keyboard rows had wrapped keys on the game pages`
  - 1080p desktop ✅
  - iPad Air ✅
  - Samsung Galaxy S20 Ultra Super Mega Excellent ❌
  - IPhone SE ❌
- Create a improved landing page and site theme (Background, make it look attractive to play) ✅ `I liked the pulsating glow on the landing page, and the background made the default theme look really good, however light mode themes suffered a bit from this`
  - Theme should extend all pages ✅ 
- Add an instructions page that is available via the hamburger menu that details the features of the game and how to play ✅
- Create a new Wordle game page that reuses components but will provide the same word for a specific day. Everyone going to the page that day will get the same word. The page should indicate Wordle of the Day or something like that ✅ `great job with the game options prop`
- Create a page that lists the last 10 daily words with statistics. Number of plays, Average score, Average time. (just list the date, don't show the word silly) ❌✅ `This mostly worked, however days that hadn't been played were missing from the list`
  - It should indicate if the user has played that day ✅
  - Clicking on the item should take the player to that game (Extra Credit) ✅
  - Create another ad page and choose one at random (Extra Credit) ✅

**Pull request closed without merging**